system/system_monitor/src/cpu_monitor/cpu_monitor_base.cpp

```diff
@@ -198,18 +243,24 @@ void CPUMonitorBase::checkUsage(diagnostic_updater::DiagnosticStatusWrapper & st
           } else {
             whole_level = std::max(whole_level, level);
           }
+          if (cpu_name != "all") cpu_usage.cpus.push_back(cpu_status);
```

Comment:
I think it is better to assign cpu_status in each case (usr, nice, etc.) and switch here.
```
if (cpu_name == "all") {
    cpu_usage.all = cpu_status;
} else {
   cpu_usage.cpus.push_back(cpu_status);
}
```

Reply:
Modified along this comment.